src/main/kotlin/com/xenomachina/argparser/Default.kt

```diff
@@ -26,6 +26,15 @@ import com.xenomachina.common.Holder
  * @param newDefault the default value for the resulting [ArgParser.Delegate]
  */
 fun <T> ArgParser.DelegateProvider<T>.default(newDefault: T): ArgParser.DelegateProvider<T> {
+    return ArgParser.DelegateProvider(ctor = ctor, defaultHolder = Holder { newDefault })
+}
+
+/**
+ * Returns a new `DelegateProvider` with the specified default value from a lambda.
+ *
+ * @param newDefault the default value for the resulting [ArgParser.Delegate]
+ */
+fun <T> ArgParser.DelegateProvider<T>.default(newDefault: () -> T): ArgParser.DelegateProvider<T> {
```

Comment:
I am a tiny bit nervous about this causing some sort of ambiguity, but I honestly can't think of a situation where a `T` and a `() -> T` could be confused.